src/App.js

```diff
@@ -3,26 +3,52 @@ import Movies from './Movies';
 import Movie from './Movie';
 import Nav from './Nav';
 import movieData from './data';
+import fetchCalls from './apiCalls'
 import './App.css';
 
 class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies,
+      movies:[],
+      singleMovieId:'',
       singleMovie:'',
+      error: '',
     }
   }
 
+componentDidMount = () => {
+  console.log('MOUNT')
+  fetchCalls.fetchData('movies')
+    .then(data => {
+      this.setState({movies: data.movies})
+    })
+    if (this.state.singleMovieId && !this.state.singleMovie) {
+      fetchCalls.fetchData(`movies/${this.state.singleMovieId}`)
+      .then(data => {
+        this.setState({singleMovie: data.movie})
+      })
```

Comment:
Be sure to use a `catch` at the end of any `.then()` chain to be able to handle unexpected errors.